src/Polly.Contrib.DuplicateRequestCollapser.RedisDistributedLock/StackExchangeRedisDistributedLockProvider.cs

```diff
@@ -0,0 +1,170 @@
+﻿#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using StackExchange.Redis;
+
+namespace Polly.Contrib.DuplicateRequestCollapser.RedisDistributedLock
+{
+    /// <summary>
+    /// Provides a lock scoped to a distributed key. Unlike other local locks, this lock provider
+    /// can throw an <see cref="OperationCanceledException"/> if the lock attempt times out.
+    /// </summary>
+    public class StackexchangeRedisDistributedLockProvider : ISyncLockProvider
```

Comment:
Do you see any value in offering users an async version?

Reply:
I do, will need to do a follow up pr after adding async lock options in the duplicate request collapser create method.